lib/matplotlib/testing/__init__.py

```diff
@@ -8,6 +8,8 @@
 import matplotlib
 from matplotlib.cbook import is_string_like, iterable
 from matplotlib import rcParams, rcdefaults, use
+# For backward compatibility
+from . import noseclasses
```

Comment:
It is needed for backward compatibility unless it wasn't there before (I actually did not check this, so it might be worth doing).
As far as I know, there is no choice other than raising a deprecation at import time for whole module deprecation.